packages/utils/src/namespaces.ts

```diff
@@ -112,6 +118,9 @@ export function buildApprovedNamespaces(
 
   const approvedNamespaces = {};
 
+  if (!Object.keys(requiredNamespaces).length && !Object.keys(optionalNamespaces).length)
```

Comment:
Could we add a context comment here similar to what you put in the PR description to clarify what is happening here and why 🙏